### ISSUE TITLE
Order of row names (gene) in the heatmap

### ISSUE THREAD

**Issue by howardhkk123** [satijalab/seurat#7974]
Dear Seurat Developers,

I am using DoHeatmap to generate a heatmap for my sc gene expression data.
I try to reorder the row names (gene symbol) based on the order in my metadata file, but the function will auto order the row names on the heatmap.
May I ask how I can restrict the auto order function in DoHeatmap?

**Comment by samuel-marsh** [comment]
Hi,

Not member of dev team but hopefully can be helpful.  What do you mean by ordering them in metadata file?  The order of genes by default is ordered by the order they are supplied to features parameter in `DoHeatmap`.

```
library(Seurat)
pbmc <- pbmc3k.SeuratData::pbmc3k.final
pbmc <- UpdateSeuratObject(pbmc)

DoHeatmap(pbmc, features = c("CD3E", "CD4", "CD8A"))

DoHeatmap(pbmc, features = rev(c("CD3E", "CD4", "CD8A")))
```

![image](https://github.com/satijalab/seurat/assets/38284410/c02125c8-e089-4c73-b6f4-fd7dbd1e1e1e)
![image](https://github.com/satijalab/seurat/assets/38284410/00373859-67b2-491e-8a51-0bfe194a9d2f)


Best,
Sam

**Comment by howardhkk123** [comment]
Hi, Sam, thx for your reply.
Sorry for the wrong statement.
Actucallt what

**Comment by howardhkk123** [comment]
Thank you for your prompt response

To clarify, what I'm aiming to do is reorder the top 30 Differentially Expressed Genes (DEGs) on the heatmap based on their p-values, with the largest p-value at the top and the smallest at the bottom. I've reordered the genes in the data frame accordingly, and now I tried to display them on the heatmap in the same order as the data frame.
For example,
```
top30 <- top30 %>% arrange(desc(top30$p_val_transform))
DoHeatmap(pbmc, features = top30$gene)
```
But it didn't work.
`DoHeatmap` will order the gene in other ways.
I was wondering if you would know how to reslove it.
